pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/DataTypeTransformer.java

```diff
@@ -151,7 +151,20 @@ static Object standardize(String column, @Nullable Object value, boolean isSingl
       }
       List<Object> standardizedValues = new ArrayList<>(numValues);
       for (Object singleValue : values) {
-        Object standardizedValue = standardize(column, singleValue, true);
+        Object standardizedValue;
+
+        // Check if the value itself is multivalued.
+        if (singleValue instanceof Object[] || singleValue instanceof List) {
+          standardizedValue = standardize(column, singleValue, false);
```

Comment:
I don't think this is correct. This can result in nested arrays being generated, and break the following part.

What you can do is to generate all standardized values for every elements, and continue when:
- There is no `Object[]`
- There is only one `Object[]`, no other value (all `null`s)

The same logic also applies to collections and map. But before getting into that, I'd invest why are we getting multiple nested arrays

Reply:
Agreed. On further inspection `ComplexTypeConfig` should be used to decipher such objects. Closing the PR since by the time the data comes to `DataTypeTransformer` it shouldn't be in this state (nested array, multi key multivalue map etc)